### ISSUE TITLE
Properties who's type implment ISet should emit uniqueItems: true

### ISSUE THREAD

**Issue by narthollis** [domaindrivendev/Swashbuckle.AspNetCore#430]
Currently Properties who's type are or implment `ISet` are emitted as arrays without `uniqueItems: true` this results in API documentation that does not properly represent the behavior of the type.

The following is a basic Schema filter to add `uniqueItems: true`  to all properties that are or implment `ISet`

```csharp
    public class HashSetSchemaFilter: ISchemaFilter
    {
        public void Apply(Schema model, SchemaFilterContext context)
        {
            if (context.SystemType.GetInterfaces().Any(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(ISet<>))
                || context.SystemType.IsGenericType && context.SystemType.GetGenericTypeDefinition() == typeof(ISet<>))
            {
                model.UniqueItems = true;
            }
        }
    }
```

**Comment by xperiandri** [comment]
@narthollis, maybe you will prepare a pull request?
@domaindrivendev, will you accept it?

**Comment by domaindrivendev** [comment]
See 0dccd09f28455d388981eedddeb080bafa756ac8. This will be generally available with the 3.0.0 release (coming soon). In the meantime, you can pull down the latest preview on myget.org to try it out.
